crates/router/src/connector/fiserv/transformers.rs

```diff
@@ -64,60 +90,74 @@ pub struct TransactionInteraction {
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for FiservPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        match item.request.payment_method_data {
-            api::PaymentMethod::Card(ref ccard) => {
-                let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
-                let amount = Amount {
-                    total: item.request.amount,
-                    currency: item.request.currency.to_string(),
-                };
+        let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
+        let amount = Amount {
+            total: item.request.amount,
+            currency: item.request.currency.to_string(),
+        };
+        let transaction_details = TransactionDetails {
+            capture_flag: Some(matches!(
+                item.request.capture_method,
+                Some(enums::CaptureMethod::Automatic) | None
+            )),
+            reversal_reason_code: None,
+        };
+        let metadata = item
+            .connector_meta_data
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
+        let session: SessionObject = metadata
+            .parse_value("SessionObject")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
 
+        let merchant_details = MerchantDetails {
+            merchant_id: auth.merchant_account,
+            terminal_id: Some(session.terminal_id),
+        };
+
+        let transaction_interaction = TransactionInteraction {
+            origin: "ECOM".to_string(), //Payment is being made in online mode, card not present
+            eci_indicator: "CHANNEL_ENCRYPTED".to_string(), // transaction encryption such as SSL/TLS, but authentication was not performed
+            pos_condition_code: "CARD_NOT_PRESENT_ECOM".to_string(), //card not present in online transaction
```

Comment:
Use enums and explain the enum variants with comments, this will be helpful in extending for other transactionInteraction types.